pepper-apis/src/main/java/org/broadinstitute/ddp/db/TransactionWrapper.java

```diff
@@ -214,17 +213,39 @@ public static synchronized TransactionWrapper getInstance(DB db) {
     }
 
     /**
-     * Re-reads the database pool configuration values in an uncached
-     * fashion.  Should only be called internally or by test code.
+     * Re-reads the database pool configuration values.
+     * @param useCache if true, pool is initialized with cached config values.  If false, config file is re-read from underlying storage.
+     * @throws InvalidConfigurationException if the configuration is invalid and cannot be used to create a new pool
      */
-    public static void reloadDbPoolConfiguration() {
+    public static synchronized void reloadDbPoolConfiguration(boolean useCache) throws InvalidConfigurationException {
```

Comment:
The password rotation test needs the ability to reload the pool using stale/cached data as well as re-reading the underlying config file.